code/io/timer.cpp

```diff
@@ -179,6 +179,39 @@ UI_TIMESTAMP ui_timestamp_delta(UI_TIMESTAMP stamp, int delta_ms)
 	return UI_TIMESTAMP(stamp.value() + delta_ms);
 }
 
+// Returns the difference between two timestamps as an int, does not check that the result will be positive.
+// Also only error check if a timestamp is invalid, but when either is immediate or never, it returns 0.
+int timestamp_get_delta(TIMESTAMP before, TIMESTAMP after)
+{
+	Assertion(before.isValid(), "timestamp_get_delta called with an invalid before timestamp%s. This is a coder mistake, please report!", (after.isValid()) ? "" : " and an invalid after timestamp.");
+	Assertion(after.isValid(), "timestamp_get_delta called with an invalid after timestamp. This is a coder mistake, please report!");
+
+
+	if (!before.isValid() || before.isImmediate() || before.isNever())
+		return 0;
+
+	if (!after.isValid() || after.isImmediate() || after.isNever())
+		return 0;	
```

Comment:
I would say that if 'before' is immediate and 'after' is never, the delta should be INT_MAX, and if reversed it should be INT_MIN.